Reportedly/Modules/Authentication/Signup/SignupViewController.swift

```diff
@@ -0,0 +1,331 @@
+    var slackIdFieldText: String? { get }
+    var isValidSignupData: Bool { get }
+    
+    func configure()
+    func setSignupButtonEnabled(_ isEnabled: Bool)
+}
+
+protocol SignupViewOutput: AnyObject {
+    
+    func viewDidLoad()
+    func didTapSignupButton()
+    func textFieldsDidChange()
+}
+
+final class SignupViewController: ViewController {
+    
+    private let backgroundImageView = UIImageView()
+    private let blurEffectView = UIVisualEffectView(effect: UIBlurEffect(style: .regular))
+    private let titleLabel = UILabel()
+    private let containerStackView = UIStackView()
+    private let emailField = UITextField()
+    private let passwordField = UITextField()
+    private let confirmPasswordField = UITextField()
+    private let slackIdField = UITextField()
+    private let signupButton = UIButton(type: .system)
+    
+    var output: SignupViewOutput?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        output?.viewDidLoad()
+    }
+    override func viewDidLayoutSubviews() {
+        emailField.roundCorners(.allCorners, radius: 8)
+        passwordField.roundCorners(.allCorners, radius: 8)
+        confirmPasswordField.roundCorners(.allCorners, radius: 8)
+        slackIdField.roundCorners(.allCorners, radius: 8)
+        signupButton.roundCorners(.allCorners, radius: 8)
+    }
+    
+    override func setUpTexts() {
+        super.setUpTexts()
+        titleLabel.text = Localize.moduleSignupHeaderTitle.localized()
+        emailField.placeholder = Localize.moduleSignupEmailPlaceholder.localized()
+        passwordField.placeholder = Localize.moduleSignupPasswordPlaceholder.localized()
+        confirmPasswordField.placeholder = Localize.moduleSignupConfirmPasswordPlaceholder.localized()
+        slackIdField.placeholder = Localize.moduleSignupSlackIdPlaceholder.localized()
+        signupButton.setTitle(Localize.moduleSignupSignupButton.localized(), for: .normal)
+    }
+    
+    override func setUpColors() {
+        super.setUpColors()
+        view.backgroundColor = .background
+        
+        titleLabel.textColor = .textPrimary
+        
+        emailField.tintColor = .primary
+        emailField.backgroundColor = .forms
+        emailField.textColor = .textPrimary
+        
+        passwordField.tintColor = .primary
+        passwordField.backgroundColor = .forms
+        passwordField.textColor = .textPrimary
+        
+        confirmPasswordField.tintColor = .primary
+        confirmPasswordField.backgroundColor = .forms
+        confirmPasswordField.textColor = .textPrimary
+        
+        slackIdField.tintColor = .primary
+        slackIdField.backgroundColor = .forms
+        slackIdField.textColor = .textPrimary
+        
+        signupButton.setTitleColor(.darkGray, for: .disabled)
+        signupButton.setTitleColor(.black, for: .normal)
+        signupButton.setBackgroundColor(.lightGray, for: .disabled)
+        signupButton.setBackgroundColor(.primary, for: .normal)
```

Comment:
Duplicated `setTitleColor ` and `setBackgroundColor ` 🙈

Reply:
oh this is intentional since I want to have different colors for different states of a button 🙏